deploy/power-agent/actuator.py

```
@@ -0,0 +1,962 @@
+                "DcgmActuator.apply_cap requires a metrics object; "
+                "construct as DcgmActuator(..., metrics=power_agent_metrics)."
+            )
+
+        # Clamp against the SKU range. constraints_w handles its own
+        # stale-handle recovery, so we don't re-wrap it here.
+        min_w, max_w = self.constraints_w(gpu_idx)
+        effective_w = self._clamp_with_metrics(watts, min_w, max_w, gpu_idx)
+
+        try:
+            return self._apply_cap_inner(gpu_idx, effective_w)
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)